src/client/Ruby/ms-rest-azure/lib/ms_rest_azure/azure_service_client.rb

```diff
@@ -14,6 +14,26 @@ class AzureServiceClient < MsRest::ServiceClient
     # @return [String] api version of the Azure in string format.
     attr_accessor :api_version
 
+    # @return [Hash{String=>String}] default request headers for Azure requests.
+    attr_accessor :request_headers
+
+    def initialize(credentials, options = nil)
+      super(credentials, options)
+      @request_headers =  {
+          'Content-Type' => 'application/json;charset=utf-8' # This is the current default for Azure services, and content-type supported by Autorest
+      }
+    end
+    #
+    # Retrieves a new instance of the AzureOperationResponse class.
+    # @param [MsRest::HttpOperationRequest] request the HTTP request object.
+    # @param [Faraday::Response] response the HTTP response object.
+    # @param [String] body the HTTP response body.
+    # @return [MsRestAzure::AzureOperationResponse] the operation response.
+    #
+    def create_response(request, http_response, body = nil)
```

Comment:
`create_response` is privately defined on base class. Silly dynamic programming language allows us everything to be re-defined :)